### PR TITLE
Set up grafana monitoring for dev

### PR DESCRIPTION
**Pull request by ethanmills**
### Context
This is a combination of stuff from https://github.com/DFE-Digital/cf-monitoring, teaching vacancies, and Get Into Teaching, tweaked for our needs. It gives a fancy grafana monitoring dashboard. I'll like to get this in to play with before doing staging and prod

### Testing
I've run this terraform locally, you should be able to sign in with google here: https://grafana-cpd-monitoring-dev.london.cloudapps.digital/?orgId=1
It lets in anyone with a digital.education.gov.uk email to view

### CODE REVIEW

File: .github/workflows/deploy_monitoring_dev.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    steps:
+      - uses: softprops/turnstyle@v1
+        name: Check workflow concurrency
+        with:
+          poll-interval-seconds: 20
+          same-branch-only: true
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  deploy:
+    name: build docker image and deploy
```
Comment (kicferk1) [inline .github/workflows/deploy_monitoring_dev.yml:19]: Are we doing anything docker related here?

**Review by cfq — APPROVED** [review]

File: .github/workflows/deploy_monitoring_dev.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+          terraform_version: 0.14.0
+
+      - name: Deploy to dev
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          TF_VAR_paas_user: ${{ secrets.GOVPAAS_DEV_USERNAME }}
+          TF_VAR_paas_password: ${{ secrets.GOVPAAS_DEV_PASSWORD }}
+          TF_VAR_google_client_id: ${{ secrets.GOOGLE_CLIENT_ID }}
+          TF_VAR_google_client_secret: ${{ secrets.GOOGLE_CLIENT_SECRET }}
```
Comment (kicferk1) [inline .github/workflows/deploy_monitoring_dev.yml:38]: What are these used for?

File: terraform/monitoring/terraform.tf [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  }
+}
+
+provider cloudfoundry {
+  api_url           = var.paas_api_url
+  password          = var.paas_password
+  sso_passcode      = ""
+  store_tokens_path = "./tokens"
+  user              = var.paas_user
+
```
Comment (kicferk1) [inline terraform/monitoring/terraform.tf:16]: Some stray whitespace here

File: terraform/monitoring/terraform.tf [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    key     = "monitoring/terraform.tfstate"
+    region  = "eu-west-2"
+    encrypt = "true"
+  }
+}
+
+provider cloudfoundry {
+  api_url           = var.paas_api_url
+  password          = var.paas_password
+  sso_passcode      = ""
```
Comment (kicferk1) [inline terraform/monitoring/terraform.tf:13]: Uhh, do we need to set this to empty string?

File: .github/workflows/deploy_monitoring_dev.yml (hunk 2 above)
Comment (ethanmills) [reply .github/workflows/deploy_monitoring_dev.yml:38]: Google SSO

File: terraform/monitoring/terraform.tf (hunk 4 above)
Comment (ethanmills) [reply terraform/monitoring/terraform.tf:13]: 🤷 let's find out

**Pull request merged**